pkgs/build-support/binary-cache/default.nix

```
@@ -0,0 +1,40 @@
+{ stdenv, buildPackages }:
+
+# This function is for creating a flat-file binary cache, i.e. the kind created by
+# nix copy --to file:///some/path and usable as a substituter (with the file:// prefix).
+
+# For example, in the Nixpkgs repo:
+# nix-build -E 'with import ./. {}; mkBinaryCache { rootPaths = [hello]; }'
+
+{ name ? "binary-cache"
+, rootPaths
+}:
+
+stdenv.mkDerivation {
+  inherit name;
+
+  __structuredAttrs = true;
+
+  exportReferencesGraph.closure = rootPaths;
+
+  preferLocalBuild = true;
+
+  PATH = "${buildPackages.coreutils}/bin:${buildPackages.jq}/bin:${buildPackages.python3}/bin:${buildPackages.nix}/bin:${buildPackages.xz}/bin";
+
+  builder = builtins.toFile "builder" ''
```

I don't think bypassing the stdenv scripts is necessary anymore since https://github.com/NixOS/nixpkgs/pull/175649 was merged. I do think we should bypass the usual phases, which is what `buildCommand` achieves.

```suggestion
  nativeBuildInputs = [ buildPackages.coreutils buildPackages.jq buildPackages.python3 buildPackages.nix buildPackages.xz ];

  buildCommand = ''
```